Dapper.MoqTests/ObjectExtensions.cs

```diff
@@ -48,15 +59,26 @@ private static DataTable GetDataTableForArray(IEnumerable value)
 
             var properties = elementType
                 .GetProperties()
-                .Where(p => IsPrimitiveType(p.PropertyType))
+                .Select(p => new { Property = p, NullablePrimitiveType = NullablePrimitiveType(p.PropertyType) })
```

Comment:
I'd suggest making a change more along the lines of:

```csharp
.Select(p => new {
  name = p.Name,
  dataType = p.PropertyType,
  nullableDataType = NullablePrimativeType(p.PropertyType) != null,
})
```
In doing so the for loop later can be simplified:

```csharp
foreach (var property in properties)
{
  var columnType = nullableDataType ?? property.PropertyType;
  var column = new DataColumn(property.name, columnType) 
  {
    AllowDBNull = nullableDataType != null
  }
  dataTable.Columns.Add(column);
}
```

Or something to that effect.